core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1281,6 +1285,9 @@ private[spark] class BlockManager(
 
     require(blockId != null, "BlockId is null")
     require(level != null && level.isValid, "StorageLevel is null or invalid")
+    if (blockManagerDecommissioning && blockId.isRDD) {
```

Comment:
I don't think we should reject the block?

Reply:
Added overall design to explain why I went ahead with this -  https://github.com/apache/spark/pull/27864#issuecomment-605599508 .

Reply:
Still not sure about this part of the design. I'd like to see a test to make sure this won't cause task failures.

Reply:
@holdenk Fixed this. Removed the code for not accepting new RDD cache blocks on Decommissioning executor. Also added a test to show that ongoing tasks which are going to cache data in future succeeds as part of BlockManagerDecommissionSuite.